js/src/Ice/OutgoingAsync.js

```diff
@@ -153,16 +147,155 @@ export class ProxyOutgoingAsyncBase extends OutgoingAsyncBase {
 
     handleException(ex) {
         const cnt = { value: this._cnt };
-        const interval = this._proxy._requestHandlerCache.handleException(
-            ex,
-            this._handler,
-            this._mode,
-            this._sent,
-            cnt,
-        );
+        const interval = this.handleRetryAfterException(ex, cnt);
         this._cnt = cnt.value;
         return interval;
```

Comment:
I'm not super familiar with JavaScript...
Do we still need to keep this weird wrapping of `this._cnt` as the field of an object?
And then update the actual field with this object's value later?

Reply:
No, you can update `this._cnt` directly in `checkRetryAfterException` now that everything is in the same class. The wrapper was required because the number type is always passed by value in JavaScript.

Reply:
Alright cool, I removed it!